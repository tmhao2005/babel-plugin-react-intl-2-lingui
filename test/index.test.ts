import { transformFileSync } from "@babel/core";
import * as fs from "fs";
import { join } from "path";
import plugin from "../src";

function trim(str?: string | null) {
  return String(str).replace(/^\s+|\s+$/, "");
}

const fixturesDir = join(__dirname, "fixtures");
const localeDir = join(__dirname, "_build/test/fixtures");

describe("fixtures", () => {
  fs.readdirSync(fixturesDir).map((caseName) => {
    // if (caseName !== 'macro-plural') return;

    it(`output match: ${caseName}`, () => {
      const fixtureDir = join(fixturesDir, caseName);
      const messagePath = join(localeDir, caseName, "actual.js.json");

      if (fs.existsSync(messagePath)) {
        fs.unlinkSync(messagePath);
      }

      const { code: actual } = transform(join(fixtureDir, "actual.js"));

      expect(trim(actual)).toMatchSnapshot();
      expect(require(messagePath)).toMatchSnapshot();
    });
  });
});

const BASE_OPTIONS = {
  localeDir: __dirname,
};

let cacheBust = 1;

function transform(filePath: string, options = {}) {
  function getPluginConfig() {
    return [
      require("@lingui/babel-plugin-extract-messages"),
      {
        ...BASE_OPTIONS,
        ...options,
      },
      Date.now() + "" + ++cacheBust,
    ];
  }

  return transformFileSync(filePath, {
    presets: ["@babel/preset-env", "@babel/preset-react"],
    plugins: [plugin, "babel-plugin-macros", getPluginConfig()],
  });
}
