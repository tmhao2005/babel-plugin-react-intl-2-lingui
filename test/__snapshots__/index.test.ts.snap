// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures output match: FormattedHTMLMessage 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.MyComponent = void 0;

var _react = require(\\"react\\");

var _react2 = require(\\"@lingui/react\\");

const MyComponent = () => {
  return (/*#__PURE__*/_react.React.createElement(_react.React.Fragment, null, /*#__PURE__*/_react.React.createElement(_react2.Trans, {
      id: \\"Hello {name}\\",
      defaults: \\"Hello {name}\\",
      values: {
        name: 'Hao'
      }
    }), /*#__PURE__*/_react.React.createElement(_react2.Trans, {
      id: \\"Online\\",
      defaults: \\"Online\\"
    }))
  );
};

exports.MyComponent = MyComponent;"
`;

exports[`fixtures output match: FormattedHTMLMessage 2`] = `
Object {
  "Hello {name}": Object {
    "defaults": "Hello {name}",
    "origin": Array [
      Array [
        "test/fixtures/FormattedHTMLMessage/actual.js",
        7,
      ],
    ],
  },
  "Online": Object {
    "defaults": "Online",
    "origin": Array [
      Array [
        "test/fixtures/FormattedHTMLMessage/actual.js",
        14,
      ],
    ],
  },
}
`;

exports[`fixtures output match: FormattedMessage 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.MyComponent = void 0;

var _react = require(\\"react\\");

var _react2 = require(\\"@lingui/react\\");

const MyComponent = () => {
  return (/*#__PURE__*/_react.React.createElement(_react.React.Fragment, null, /*#__PURE__*/_react.React.createElement(_react2.Trans, {
      id: \\"Hello\\",
      defaults: \\"Hello\\"
    }), /*#__PURE__*/_react.React.createElement(_react2.Trans, {
      id: \\"app.name\\",
      defaults: \\"You\\"
    }))
  );
};

exports.MyComponent = MyComponent;"
`;

exports[`fixtures output match: FormattedMessage 2`] = `
Object {
  "Hello": Object {
    "defaults": "Hello",
    "origin": Array [
      Array [
        "test/fixtures/FormattedMessage/actual.js",
        8,
      ],
    ],
  },
  "app.name": Object {
    "defaults": "You",
    "origin": Array [
      Array [
        "test/fixtures/FormattedMessage/actual.js",
        9,
      ],
    ],
  },
}
`;
